terraform/modules/compute/wls_compute/wls_volume.tf

```diff
@@ -18,7 +18,7 @@ module "middleware-volume" {
 
 module "data-volume" {
   source = "../volume"
-  bv_params = { for x in range(var.num_vm_instances) : "${var.resource_name_prefix}-data-block-${x}" => {
+  bv_params = { for x in range(var.num_vm_instances) : "${var.resource_name_prefix}-data-block-${format("%02d", x)}" => {
     ad             = var.use_regional_subnet ? local.ad_names[(x + local.admin_ad_index) % length(local.ad_names)] : var.availability_domain
     compartment_id = var.compartment_id
     display_name   = "${var.resource_name_prefix}-data-block-${x}"
```

Comment:
The change looks good to me.

nitpick: 

I noticed you chose to keep the display name single digit (for example -  display name is "-1" for data-block "-01").  This is not a functional problem.

Reply:
Good observation. It was made on purpose, I wanted to make the least possible changes. There might be code that checks the server names, VM display names, etc. I am not sure and I do not want to break such code, if any
The key of the maps used to create the resource collections are not used in other code that is not terraform, that I am sure. That